app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all.order(:id)
+  end
+
+  def show 
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+
+  def new 
+    @task = Task.new
+    @task.task = "I have to... "
+    @task.completed = "Incomplete"
+  end
+
+  def create 
+    new_task = Task.new(task_params)
+
+    is_successful = new_task.save
+
+    if is_successful
+      redirect_to task_path(new_task.id)
+    else
+      head :not_found
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+
+  def update
+    existing_task = Task.find_by(id: params[:id])
+
+    # is_successful = existing_task.update(task_params)
+
+    if existing_task && existing_task.update(task_params)
+      redirect_to task_path(existing_task.id)
+    else
+      redirect_to tasks_path
+    end
+  end
+
+  def complete
+    task = Task.find_by(id: params[:id])
+
+    task.completed?
```

Comment:
The line `task.completed?` does nothing -- It just returns a boolean and then does nothing with it. Usually, we pair booleans with conditionals. What were you intending to do with this line of code?